### PR TITLE
Modselect refactor

### PR DESCRIPTION
**Pull request by huoyaoyuan**
Depends on #448.

### CODE REVIEW

**Comment by peppy** [comment]
tom and i decided we shouldn't be using `new` for things like Colour to avoid confusion. I'd recommend renaming your colour to `ButtonColour` or similar.

**Comment by huoyaoyuan** [comment]
Questions: 
  - Where to provide hotkey of mods?
    - `ModSection`, determined with layout
    - by every mod itself
  - Allow null to generate a placeholder, or not?
    - if not, I will revert the last 2 commits

**Comment by peppy** [comment]
What is the advantage of making placeholder icons?

File: osu.Game/Overlays/Mods/DifficultyIncreaseSection.cs [hunk 1]
```diff
@@ -4,17 +4,19 @@
...
     {
-        protected override Key[] ToggleKeys => new[] { Key.A, Key.S, Key.D, Key.F, Key.G, Key.H, Key.J, Key.K, Key.L };
+        protected override Key[] ToggleKeys => new Key[] { Key.A, Key.S, Key.D, Key.F, Key.G, Key.H, Key.J, Key.K, Key.L };
+        public override ModType ModType => ModType.DifficultyIncrease;
 
         [BackgroundDependencyLoader]
         private void load(OsuColour colours)
         {
-            Colour = colours.Yellow;
+            BottonColour = colours.Yellow;
```
Comment (peppy) [inline osu.Game/Overlays/Mods/DifficultyIncreaseSection.cs]: Button

**Comment by peppy** [comment]
Please merge master in.

**Comment by peppy** [comment]
oh. master is merged in.

**Pull request merged**